diff --git a/dist/runtime/utils/renderer/build-files.mjs b/dist/runtime/utils/renderer/build-files.mjs
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/runtime/utils/renderer/build-files.mjs
+++ b/dist/runtime/utils/renderer/build-files.mjs
@@ -7,7 +7,11 @@ import { NUXT_NO_SSR } from "#internal/nuxt/nitro-config.mjs";
 // @ts-expect-error virtual file
 import { appRootAttrs, appRootTag, appSpaLoaderAttrs, appSpaLoaderTag, spaLoadingTemplateOutside } from "#internal/nuxt.config.mjs";
 // @ts-expect-error virtual file
-import { buildAssetsURL } from "#internal/nuxt/paths";
+import { buildAssetsURL, publicAssetsURL } from "#internal/nuxt/paths";
+// @ts-expect-error private property consumed by vite-generated url helpers
+globalThis.__buildAssetsURL = buildAssetsURL;
+// @ts-expect-error private property consumed by vite-generated url helpers
+globalThis.__publicAssetsURL = publicAssetsURL;
 const APP_ROOT_OPEN_TAG = `<${appRootTag}${propsToString(appRootAttrs)}>`;
 const APP_ROOT_CLOSE_TAG = `</${appRootTag}>`;
 // @ts-expect-error file will be produced after app build
